### PR TITLE
MTU Override & Raise Options

### PR DESCRIPTION
**Pull request by Moe-Baker**
Adding options to override & raise MTU, the override will ignore MTU discovery while the raise will only raise the MTU one rank to 1232 - MaxUdpHeaderSize.
Thought these would be helpful since the MTU gets set to a very low minimum of ~500B before MTU discovery.

### CODE REVIEW

**Comment by aienabled** [comment]
I'm glad to see people got thinking after my post in Discord (I use @VK nickname there).
However, my idea was that the auto-discovery should still be used, it's just that the minimum MTU by default should be something like 1024 (a practical lower bound instead of the theoretical one).
MTU discovery is very helpful as it reduces fragmentation automatically (for reliable packets) and for unreliable packets there is an API to get the current max payload size so the library user can maximize their unreliable packet payload according to the current MTU.

Regards!

**Comment by aienabled** [comment]
I'm glad to see people got thinking after my post in Discord (I use @VK nickname there).
However, my idea was that the auto-discovery should still be used, it's just that the minimum MTU by default should be something like 1024 (a practical lower bound instead of the theoretical one).
MTU discovery is very helpful as it reduces fragmentation automatically (for reliable packets) and for unreliable packets there is an API to get the current max payload size so the library user can maximize their unreliable packet payload according to the current MTU.

TL/DR: instead of simply overriding MTU, it's best to 1) provide an API to configure the minimum MTU for MTU path discovery (so it could be used even in a NuGet version of the library) and 2) to make the minimum MTU higher.

Regards!

**Comment by Moe-Baker** [comment]
> However, my idea was that the auto-discovery should still be used, it's just that the minimum MTU by default should be something like 1024

only the override will completely cancel discovery, the raise method should up the MTU to 1,232 bytes and continue with MTU discovery from there, tho one issue is that MTU will only go up or stay at 1,232 so if you do end up with one of those unicorn players who do indeed have a 500 MTU ... well, I don't know what will happen but I imagine it won't be pretty for them.

**Comment by aienabled** [comment]
@Moe-Baker thank you for a quick response!
I've checked the diff now completely. I don't like the particular API (`RaiseInitialMTU`) but yes, it does the job!

My suggestion would be to introduce a property akin to other settings, as it makes perfect sense to make it a setting, not a method. Something like `UseLowMinimumMtu` (which is best to keep `false` by default as I've explained before) or maybe just an ushort property `MinimumPossibleMtu` that changes the minimum possible MTU for Path MTU discovery procedure (I can see that it would be much harder to implement though, considering the current architecture). That would allow better customization options without the need to edit the source code.

I would also suggest expanding the `NetConstants.PossibleMtu` array with more options (in the range between 1024 and 1232) as the current set is quite restrictive. With 1024 bytes it's already way more headroom for the payload than with 576 bytes and as I've mentioned in my Discord post, 1024 is also used in Unreal Engine 4 (would be great if somebody will make further research by checking other libraries and games, and, perhaps, contacting programmers from AAA studios (developers of Fortnite, PUBG, Ark, Conan, other popular games) to learn about their approach firsthand).

Regards!

**Comment by Moe-Baker** [comment]
Yeah, that would be optimal, yet the current solution fits my needs tho so I'm good, lol.
Hopefully someone can take it further from here.

**Comment by RevenantX** [comment]
"someone"
